### PR TITLE
fix 2 connection divide by zero errors

### LINKED ISSUE #58: Divide by 0 bug
A divide by 0 somewhere in the canvas connection layout logic causes connections to be displayed incorrectly in some corner cases.
Example:
```
classDiagram {
    class("Test") layout {
        pos = apos(0, -140)
    }

    class("Test2") layout {
        pos = apos(0, 260)
    }

    Test --> Test2 with {
        over = start(0.25).axisAligned(0, end(0.75))
    }
}
```
image

### PR DESCRIPTION
**Pull request by nk-coding**
## bugfixes
- fixes marker (and path) layouting, if
  - a axis-aligned segment is a vertical/horizontal line
  - any type of segment starts where it ends
    - in this case a rotation for the marker cannot be calculated, thus defaults to 0
    - also this is a dumb corner case which should never happen in practice for any valid diagram

## issues
- closes #58

### CODE REVIEW

File: packages/diagram-common/src/model/elements/canvas/canvasAxisAlignedSegment.ts [hunk 1]
```diff
@@ -59,10 +59,15 @@ export namespace CanvasAxisAlignedSegment {
...
+        let helperPos: Point;
         if (pos > -1 && pos <= 0) {
-            return calculateMarkerRenderInformationInternal(point, { x: point.x, y: endPoint.y }, marker);
+            helperPos = { x: point.x, y: endPoint.y };
         } else {
-            return calculateMarkerRenderInformationInternal(point, { x: endPoint.x, y: point.y }, marker);
+            helperPos = { x: endPoint.x, y: point.y };
         }
+        if (Point.equals(helperPos, point)) {
+            helperPos = endPoint;
```
Comment (delvh) [inline packages/diagram-common/src/model/elements/canvas/canvasAxisAlignedSegment.ts:69]: Ah, is that the bug we observed yesterday together?

**Review by delvh — APPROVED** [review]

File: packages/diagram-common/src/model/elements/canvas/canvasAxisAlignedSegment.ts (hunk 1 above)
Comment (nk-coding) [reply packages/diagram-common/src/model/elements/canvas/canvasAxisAlignedSegment.ts:69]: yes
